### ISSUE TITLE
Exchanges' required credentials and metainfo

### ISSUE THREAD

**Issue by MaximKorobov** [ccxt/ccxt#541]
I should ask user for exchanges requisites by UI.
How to get what requisites are required to exact exchange? For example, that Exmo and LiveCoin need apiKey + secret, bitlish needs only apiKey?

**Comment by kroitor** [comment]
Hi, @MaximKorobov !

Thanks a lot for opening this issue! Currently, there is no way to tell that programmatically from the library, apart from hardcoding those cases in the userland. As you may know, this library is under active development, so not all wanted features are implemented yet, however, this particular feature is on our roadmap. And we hope to deliver it asap as this functionality is needed by most of users. 

We also welcome your contributions and PRs, so, if you feel like you want to give it a try and implement this feature - go ahead, please, we will be thankful for your contributions and PRs (read more: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md).

Thx again!

**Comment by MaximKorobov** [comment]
As far authentication parameters are almost already in library code (most in `sign` method of Exchanges) please consider to declare such meta explicitly.

**Comment by kroitor** [comment]
@MaximKorobov sure, will add those properties to the exchange asap.

**Comment by kroitor** [comment]
Ok, now every exchange has the following property:

```
exchange.requiredCredentials = {
    'apiKey': true,
    'secret': true,
    'uid': false,
    'login': false,
    'password': false,
}
```

It will throw an AuthenticationError if you try to access their authenticated API while missing any of `true` values on the exchange instance. Some other keys may be added in future as well. The above functionality is available in ccxt as of version 1.10.175. Let us know if you have any difficulties with it. 

Thanks again for the feedback! Cheers!
